trading/models.py

```diff
@@ -38,15 +38,17 @@ def available_cash(self):
         ])
         return stock_val + bucket_val
 
-    @staticmethod
-    def available_buckets(bkt):
+    def available_buckets(self, bkt):
         """
         Find the available buckets that have quantity > 0
         """
-        bucket = TradeBucket.buckettrades.filter(bucket=bkt).annotate(
+        bucket = self.buckettrades.all().filter(stock=bkt.id).annotate(
             sum_quantity=models.Sum('quantity'))
```

Comment:
Use get instead of all, since we can have at most one after aggregation

Reply:
Wrap the get in a try catch (Python EAFP)

Reply:
But doesn't the all have to evaluate before the aggregation, so there's something to aggregate on?